packages/react-query/src/utils/inferReactQueryProcedure.ts

```diff
@@ -30,7 +30,7 @@ export type InferQueryOptions<
     TRPCClientErrorLike<TRoot>,
     TData
   >,
-  'select'
+  'select' | 'queryFn'
```

Comment:
cc @Nick-Lucas - this seems to do it but I'm ngl I don't know if there're broader implications by this change 😅